### PR TITLE
Fix unmount iso/virtual drives after eject

### LINKED ISSUE #6072: Ejecting ISOs does not unmount the virtual drive.
### Description

If you eject an ISO in Windows Explorer, it will also remove the virtual drive. If you do it in Files, it won't remove the virtual drive. If you look in Explorer after ejecting the ISO, you will see an empty virtual drive.

### Steps To Reproduce

1. Go to the home page with the drives.
2. Eject the ISO.
3. View the drives in Explorer.

### Expected behavior

The virtual drive should also be ejected along with the ISO.

### Files Version

v2.0.11.0

### Windows Version

Windows 11 Home 21H2 22000.176

### Relevant Assets

_No response_

### PR DESCRIPTION
**Pull request by DavidPerikala**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #6072

**Details of Changes**
Add details of changes here.
- Fixed Unmounting ISO files to remove label after ejecting.
- I could see that the case for ejecting is same as inserting? How? I guess it should be the same as Removing, right? Or is there a reason for that? 
- I updated and tested it. It works. 

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

**Comment by gave92** [comment]
I think it was like that because when you have a "real" CDrom drive, when you eject a CD you don't want to remove the drive from the list.

**Comment by gave92** [comment]
Plus if you do this I think the empty drive still appears in windows explorer.

**Comment by DavidPerikala** [comment]
> I think it was like that because when you have a "real" CDrom drive (or even a daemon tools-like drive), when you eject a CD you don't want to remove the drive from the list, but just update the label.

Oh okay. Now I understand. 



> Plus even if you do this I think the empty drive will still appear in windows explorer.

oops yes. I see it :/
Any hint, how we can proceed here?

**Comment by gave92** [comment]
Uh good question. I guess we could:
- check what happens if we can invoke the "Eject" verb of the explorer context menu from Files
- check if dismounting the iso from powershell removes the drive from explorer (e.g. with `Dismount-DiskImage -ImagePath "E:\Windows10.iso"`)

**Comment by gave92** [comment]
Just checked [2]:
Calling e.g. `Dismount-DiskImage -ImagePath "E:\Windows10.iso"` will also remove the drive from explorer.
Guess this issue is the same as #6439 (for removable drives), we do not Eject the device/ISO correctly.

**Comment by gave92** [comment]
Update: pushed a change to eject using shell. Appears to fix both #6072 and #6439

**Comment by DavidPerikala** [comment]
Wow. I was looking in a different implementation but great, Thank you @gave92

**Comment by gave92** [comment]
Thanks! Does it work on your end as well then?
I'm not super happy with this solution, I was hoping to accomplish this in UWP without depending on file explorer.
So if you've found a different solution feel free to revert my changes.

**Comment by DavidPerikala** [comment]
Yes I tried and it works. 
It's ok. I guess we can use this fix for now?. But I will try an implementation and send a new PR for that it that works.

**Comment by DavidPerikala** [comment]
@gave92 Hold on! It doesn't remove the pendrive. so #6439 is still reproducible. Can you verify once?

**Comment by gave92** [comment]
I can confirm that on my end removing a usb key works (I have only one with which to try though). I'm on windows 10 (if that matters),
Do you get any error message? Do you get the "Safe to remove Hardware" windows notification?

**Comment by DavidPerikala** [comment]
I am on win 11 & So there's safe to remove hardware notification & no error. But the drive is still there on the UI. Once I remove it physically it's removed. Debug log doesn't show any error.

**Comment by gave92** [comment]
Does the drive appear in Explorer after you eject it from Files?

**Comment by DavidPerikala** [comment]
> Does the drive appear in Explorer after you eject it from Files (before you remove it physically)?

No, It's gone from the Explorer. (before removing physically)



> Does the drive disappear from FIles if you use the code from `main` (without this PR)?

No, it doesn't disappear from Files on the main branch

**Comment by gave92** [comment]
> No, it doesn't disappear from Files on the main branch

Thanks for confirming. If this PR does break anything I think we can merge it as it solves at least #6920.

File: src/Files.App/ViewModels/ItemViewModel.cs [hunk 1]
```diff
@@ -407,6 +407,8 @@ public ItemViewModel(FolderSettingsViewModel folderSettingsViewModel)
             UserSettingsService.OnSettingChangedEvent += UserSettingsService_OnSettingChangedEvent;
             FileTagsSettingsService.OnSettingImportedEvent += FileTagsSettingsService_OnSettingImportedEvent;
             FolderSizeProvider.SizeChanged += FolderSizeProvider_SizeChanged;
+
+            _ = InitializeConnectionAsync(); // fire and forget
```
Comment (d2dyno1) [inline src/Files.App/ViewModels/ItemViewModel.cs:411]: I don't think it should belong in the constructor

Comment (gave92) [reply src/Files.App/ViewModels/ItemViewModel.cs]: Copied from YourHomeViewModel. Moved outside the constructor now.

**Review by gave92 — APPROVED** [review]

**Pull request merged**